BitwardenShared/Core/Platform/Services/ConfigService.swift

```diff
@@ -157,6 +164,22 @@ class DefaultConfigService: ConfigService {
 
     // MARK: Private
 
+    /// Loads the flags into the SDK.
+    /// - Parameter config: Config to update the flags.
+    private func loadFlags(_ config: ServerConfig?) async {
+        do {
+            guard let config else {
+                return
+            }
+
+            try await clientService.platform().loadFlags([
+                FeatureFlagsConstants.enableCipherKeyEncryption: config.supportsCipherKeyEncryption(),
+            ])
```

Comment:
⚠️ One thing I'm noticing is that this doesn't seem to get called when a new user signs into the app. I think the config is fetched for that user's environment prior to login, but since we don't have a user ID for the account yet to save that config, it's not saved to user defaults nor is it loaded here.

⚠️ And a related issue, since we rely on the active user ID when saving the config, I think there's potential for us to be saving a config to the wrong user account. For example, if you have an existing account signed in (userId=1) and you go to add an account in a different environment (userId=2), we load the config for that environment and then save it for the active user id (userId=1) since the new user isn't the active user yet.

@fedemkr Do you know offhand how MAUI handles this?

Neither of these were introduced here, so this work could be done separately but I'm not sure how it impacts this.

One potential idea: `ConfigService` could load the config for an environment and save it as a pre-auth config, similar to pre-auth URLs. And then when the user logs in, we save this pre-auth config to that user's config. We'd just need a way to indicate whether we're fetching the config for the active user or a potential new account.

Reply:
We might have a similar issue on MAUI. The main difference is that on MAUI we're not saving a config per user but we maintain one globally which is fetched again when there's a change in user. So I think the native app does a much better job in this matter.
Yes, I like your idea of having a `pre-auth` config or something like that but I've just thought of another solution.

💡 @matt-livefront  What do you think about using the server url as key instead of a user id? We can use a hash of the server url if needed. With this we avoid having to worry about `pre-auth` or things like that. Additionally, we can reuse the same config amongst users of the same server.

Reply:
> 💡 @matt-livefront What do you think about using the api server url as key instead of a user id? We can use a hash of the server url if needed. With this we avoid having to worry about pre-auth or things like that. Additionally, we can reuse the same config amongst users of the same server.

Great idea, I like this approach a lot.

> As to this not being called when the user logs in, I'll make some changes for that to happen. Maybe call getting the config every time the active user changes 🤔

This makes sense to me!